### PR TITLE
Add Starksheet, Kakarot and Kakasheet to the stack

### PR DESCRIPTION
**Pull request by ClementWalter**
# Add Starksheet, Kakarot, Kakasheet and Madara swap

Also removed the unused misc-apps

### CODE REVIEW

**Comment by d-roak** [comment]
I'm finishing the healthcheck directly on Dockerfile, let's merge this one after. Any reason to remove completely misc-apps? I would like to keep madara-swap and kakarot deployment scripts while we don't have an alternative

**Comment by ClementWalter** [comment]
> I'm finishing the healthcheck directly on Dockerfile, let's merge this one after. Any reason to remove completely misc-apps? I would like to keep madara-swap and kakarot deployment scripts while we don't have an alternative

I've updated the PR with the full kakarot stack (contract + RPC)

I can add a madara-swap deployment as well with the same strategy (build a deployer image in the madara-swap repo and just use it here). I don't think that these misc-apps (that I did) were a good strategy

**Comment by ClementWalter** [comment]
lol now that I've put kakarot RPC, I could also add kakasheet (starksheet evm in kakarot)

File: starknet-stack/docker-compose.yml [hunk 1]
```diff
@@ -29,39 +33,158 @@ services:
...
+      - internal
+    # Hack, remove when kakarot-rpc adds a HEALTHCHECK
+    healthcheck:
+      test: ["CMD", "echo", "ok"]
+      interval: 5s
+
+  kakasheet-deployer:
+    image: evmsheet-deployer
+    # image: ghcr.io/the-candy-shop/starksheet-monorepo/starksheet-solidity:latest
+    platform: linux/x86_64
```
Comment (d-roak) [inline starknet-stack/docker-compose.yml:144]: this requires an emulated environment in some devices. Can we have an alternative?

**Review by d-roak — CHANGES_REQUESTED** [review]

File: starknet-stack/docker-compose.yml (hunk 1 above)
Comment (d-roak) [reply starknet-stack/docker-compose.yml:144]: that image is also not on a registry nor being built

File: starknet-stack/docker-compose.yml [hunk 2]
```diff
@@ -13,6 +13,10 @@ services:
       - "--dev"
     networks:
       - internal
+    # Hack, remove when madara adds a HEALTHCHECK
+    healthcheck:
```
[the remark this answers is not in the document]
Comment (ClementWalter) [reply starknet-stack/docker-compose.yml:17]: Actually I tried but it still doesn't work, see corresponding `docker compose ps` commands
- without

```
docker compose ps
NAME                      IMAGE                                         COMMAND                  SERVICE             CREATED             STATUS              PORTS
starknet-stack-madara-1   ghcr.io/keep-starknet-strange/madara:latest   "/madara-bin --rpc-e…"   madara              38 seconds ago      Up 38 seconds       0.0.0.0:9615->9615/tcp, 0.0.0.0:9944->9944/tcp, 0.0.0.0:30333->30333/tcp
```

- with
```
docker compose ps
NAME                                IMAGE                                         COMMAND                  SERVICE             CREATED             STATUS                   PORTS
starknet-stack-kakarot-deployer-1   ghcr.io/kkrt-labs/kakarot/deployer:latest     "python scripts/depl…"   kakarot-deployer    7 seconds ago       Up Less than a second    
starknet-stack-madara-1             ghcr.io/keep-starknet-strange/madara:latest   "/madara-bin --rpc-e…"   madara              7 seconds ago       Up 6 seconds (healthy)   0.0.0.0:9615->9615/tcp, 0.0.0.0:9944->9944/tcp, 0.0.0.0:30333->30333/tcp
```

[the remark this answers is not in the document]
Comment (ClementWalter) [reply starknet-stack/docker-compose.yml:17]: I just pulled latest, and this makes me use image `sha256:be68c3bfe68a696571bf9b41287b0f8e8f003a9a7cfe6a05da2382a39a3c1c97`

File: starknet-stack/docker-compose.yml (hunk 1 above)
Comment (ClementWalter) [reply starknet-stack/docker-compose.yml:144]: sorry this is a mistake, forgot to update from another docker compose, the build image in ghcr is multi arch

File: starknet-stack/docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (d-roak) [reply starknet-stack/docker-compose.yml:17]: let's try once this finishes: https://github.com/keep-starknet-strange/madara/actions/runs/5677451093

**Review by d-roak — APPROVED** [review]
가보자고

**Pull request merged**